crates/net/discv5/src/config.rs

```diff
@@ -286,12 +312,104 @@ impl Config {
 
     /// Returns the RLPx (TCP) socket contained in the [`discv5::Config`]. This socket will be
     /// advertised to peers in the local [`Enr`](discv5::enr::Enr).
-    pub fn rlpx_socket(&self) -> SocketAddr {
-        let port = self.tcp_port;
-        match self.discv5_config.listen_config {
-            ListenConfig::Ipv4 { ip, .. } => (ip, port).into(),
-            ListenConfig::Ipv6 { ip, .. } => (ip, port).into(),
-            ListenConfig::DualStack { ipv4, .. } => (ipv4, port).into(),
+    pub fn rlpx_socket(&self) -> &SocketAddr {
+        &self.tcp_socket
+    }
+}
+
+/// Returns the IPv4 discovery socket if one is configured.
+pub fn ipv4(listen_config: &ListenConfig) -> Option<SocketAddrV4> {
```

Comment:
not great that we have to add functions for this, but doesn't really matter imo, because not worth it rolling a wrapper type

Reply:
in my backlog to upstream ;)